### ISSUE TITLE
When use  activation='tanh' , the training program will crash

### ISSUE THREAD

**Issue by wenfengyu** [tensorflow/tensorflow#46028]
def make_generator_model():
    model = tf.keras.Sequential()
    model.add(layers.Dense(7*7*256, use_bias=False, input_shape=(100,)))
    model.add(layers.BatchNormalization())
    model.add(layers.LeakyReLU())

    model.add(layers.Reshape((7, 7, 256)))
    assert model.output_shape == (None, 7, 7, 256) 

    model.add(layers.Conv2DTranspose(128, (5, 5), strides=(1, 1), padding='same', use_bias=False))
    assert model.output_shape == (None, 7, 7, 128)
    model.add(layers.BatchNormalization())
    model.add(layers.LeakyReLU())

    model.add(layers.Conv2DTranspose(64, (5, 5), strides=(2, 2), padding='same', use_bias=False))
    assert model.output_shape == (None, 14, 14, 64)
    model.add(layers.BatchNormalization())
    model.add(layers.LeakyReLU())

    model.add(layers.Conv2DTranspose(1, (5, 5), strides=(2, 2), padding='same', use_bias=False, activation='tanh'))
    model.add(layers.LeakyReLU())
    assert model.output_shape == (None, 28, 28, 1)

    return model

generator =make_generator_model()

noise = tf.random.normal([1, 100])
generated_image = generator(noise, training=False)

###################
issue:  when use activation='tanh' , the training program will crash, use other activation function is normal.
tensorflow version:   from tf-nightly2.5.0.dev20201217 to tf-nightly2.5.0.dev20201228
cuda version:  11.0
python version: 3.8
os:  windows 10

**Comment by wenfengyu** [comment]
python version: from 3.6 to 3.8 have been tested, it will crash.

**Comment by wenfengyu** [comment]
GPU: NVIDIA RTX3070

**Comment by wenfengyu** [comment]
The test code that found this issue:   https://www.tensorflow.org/tutorials/generative/dcgan

**Comment by Saduf2019** [comment]
@wenfengyu 
I ran the code on [tf 2.4](https://colab.research.google.com/gist/Saduf2019/311112af97722381493224772d765db8/untitled488.ipynb) and nightly, i do not face any issues please find the [gist here](https://colab.research.google.com/gist/Saduf2019/e7f5788bccd563812d750f61f9ae88c8/untitled489.ipynb).

Please ensure there are no other processes running that could consume memory, due to which you could be facing a crash.

**Comment by wenfengyu** [comment]
i checked other processes an memory, it's normal.  this issue only exists on windows, nightly==2.5.0

**Comment by wenfengyu** [comment]
i checked other processes and memory, it's normal. this issue only exists on windows, nightly==2.5.0

**Comment by Saduf2019** [comment]
@wenfengyu 
Can you try the 2.4 stable version and let us know, if you face this issue.

**Comment by wenfengyu** [comment]
After installing tensorflow2.4rc4 version on windows10, cudnn initialization error appears when running the training code.
error info: Unimplemented: kernel reported driver version not implemented on Windows

tensorflow version: tensorflow 2.4.0rc4
os : windows 10
cuda version: 11.0
python version: 3.8
cudnn: version: 8.0.4
GPU: RTX3070

2020-12-29 18:23:45.222526: E tensorflow/stream_executor/cuda/cuda_dnn.cc:336] Could not create cudnn handle: CUDNN_STATUS_NOT_INITIALIZED
2020-12-29 18:23:45.222972: E tensorflow/stream_executor/cuda/cuda_dnn.cc:340] Error retrieving driver version: Unimplemented: kernel reported driver version not implemented on Windows
2020-12-29 18:23:45.225297: E tensorflow/stream_executor/cuda/cuda_dnn.cc:336] Could not create cudnn handle: CUDNN_STATUS_NOT_INITIALIZED
2020-12-29 18:23:45.225401: E tensorflow/stream_executor/cuda/cuda_dnn.cc:340] Error retrieving driver version: Unimplemented: kernel reported driver version not implemented on Windows
2020-12-29 18:23:45.225508: F tensorflow/core/kernels/conv_grad_input_ops.cc:1173] Check failed: stream->parent()->GetConvolveBackwardDataAlgorithms( conv_parameters.ShouldIncludeWinogradNonfusedAlgo(stream->parent()), &algorithms)

**Comment by ymodak** [comment]
Can you try upgrading your nvidia gpu driver version?
> Unimplemented: kernel reported driver version not implemented on Windows

After that you may reboot your system and try [limiting gpu memory growth](https://www.tensorflow.org/guide/gpu#limiting_gpu_memory_growth).
```python
# On top of your script 
import tensorflow as tf
gpus = tf.config.experimental.list_physical_devices('GPU')
tf.config.experimental.set_memory_growth(gpus[0], True)
# Rest of your code
...

**Comment by wenfengyu** [comment]
the tensorflow 2.4.0rc4 error of Could not create cudnn handle: CUDNN_STATUS_NOT_INITIALIZED  already solved by limit gpu memory growth.  but tensorflow 2.4.0rc4 can't use activation='tanh' , it will be no gradient drop in training process. Now I solve this problem by replacing activation='tanh' with a custom tanh function.
